src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2419,6 +2419,13 @@ public ObjectId getHeadRev(String url, String branchSpec) throws GitException, I
     /** {@inheritDoc} */
     public Map<String, ObjectId> getRemoteReferences(String url, String pattern, boolean headsOnly, boolean tagsOnly)
             throws GitException, InterruptedException {
+        return getRemoteReferences(url, pattern != null ? Collections.singletonList(pattern) : Collections.EMPTY_LIST,
+                headsOnly, tagsOnly);
+    }
+
+    /** {@inheritDoc} */
```

Comment:
Who knows, maybe you created this rule in ancient java times and there no other ways. Thanks for description, now i know source of this inheritDocs.